src/complex_tensor/ops/_common.py

```diff
@@ -131,7 +132,7 @@ def ordered_impl(*args, **kwargs):
 
 def register_binary_nonlinear(op: OpType) -> Callable:
     def impl(lhs: ComplexTensor, rhs: ComplexTensor, *args, **kwargs) -> ComplexTensor:
-        a_r, a_i = split_complex_tensor(lhs)
+        a_r, a_i = split_complex_arg(lhs)
```

Comment:
This is needed as any of `lhs` and `rhs` may be a `ComplexTensor`, but it isn't guaranteed which one.